### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by PierreMarTich**
Added 4 Python scripts : 
-CollectTool corresponds in the first required task, which is collecting the frames from several videos
-ExtractMarkup extracts the marked up areas using the csv file extracted from VGG
-Feature Extraction puts the images in the format readable for pix2pix (input - output). 3 features are available : edges, greyscale, and pair 
-ImageToCSVTool which converts a folder of images into a csv file.

### CODE REVIEW

File: CollectTool.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import cv2
+import os
+
+#class CollectTool:
```
Comment (licensedtroubleshooter) [inline CollectTool.py]: Why did you remove class from here?

File: CollectTool.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import cv2
+import os
+
+#class CollectTool:
+
+def captureFramesFromVideo(inputPath, outputPath):
+# Captures the frames of videos located in the inputPath and stores them in the outputPath. By default, outputPath=inputPath
+# The fps parameter is the frequency of frame capture per second, by default 2.
+    if outputPath =="":
+        outputPath=inputPath
```
Comment (licensedtroubleshooter) [inline CollectTool.py]: binary operators in python should be separated with spaces from both sides (according to PEP8- https://www.python.org/dev/peps/pep-0008/)

File: CollectTool.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# The fps parameter is the frequency of frame capture per second, by default 2.
+    if outputPath =="":
+        outputPath=inputPath
+    vidcap = cv2.VideoCapture(inputPath)
+    success, image = vidcap.read()
+    count = 0
+    success = True
+    while success:
+        success, image = vidcap.read()
+        cv2.imwrite(outputPath+"/"+"frame%d.jpg" % count, image)  # save frame as JPEG file
```
Comment (licensedtroubleshooter) [inline CollectTool.py]: same

File: CollectTool.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import cv2
+import os
+
+#class CollectTool:
+
+def captureFramesFromVideo(inputPath, outputPath):
+# Captures the frames of videos located in the inputPath and stores them in the outputPath. By default, outputPath=inputPath
```
Comment (licensedtroubleshooter) [inline CollectTool.py]: you better put a comments before function

File: CollectTool.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import cv2
+import os
+
+#class CollectTool:
+
+def captureFramesFromVideo(inputPath, outputPath):
+# Captures the frames of videos located in the inputPath and stores them in the outputPath. By default, outputPath=inputPath
+# The fps parameter is the frequency of frame capture per second, by default 2.
```
Comment (licensedtroubleshooter) [inline CollectTool.py]: Maybe we can add another function argument 'fps' so we can control it?

File: CollectTool.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        success, image = vidcap.read()
+        cv2.imwrite(outputPath+"/"+"frame%d.jpg" % count, image)  # save frame as JPEG file
+        count += 1
+
+def allVids(inputPath, outputPath):
+# Captures frames from all the videos in a folder using captureFramesfromVideo
+    if outputPath=="":
+        outputPath=inputPath
+    for file in os.listdir(inputPath):
+        filename=file[:-4]
```
Comment (licensedtroubleshooter) [inline CollectTool.py]: Remember what've told you about 'magic numbers'?

File: ExtractMarkupTool.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+    # Imports the .csv file extracted from VGG Image Annotator. Works for sure with only one region per frame.
+    # Unfortunately, there are a lots of syntaxes that I could not use without deleting them to make the data usable
+    def import_csv(self):
+        data = pd.read_csv(self.csv_path)
+
+        # Deletes images that are not labelled in the .csv file
+        data = data[data.values[:, CSV_COLUMN_INDEX_REGION_SHAPE_ATTRIBUTES] != "{}"]
+
+        data2 = pd.DataFrame(data.values[:, CSV_COLUMN_INDEX_REGION_SHAPE_ATTRIBUTES],
```
Comment (licensedtroubleshooter) [inline ExtractMarkupTool.py]: how about better naming? or just reassign 'data'?

File: ExtractMarkupTool.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        for element in x:
+            xList.append(element.split(",", ))
+
+        # Converting them to integers
+        for element in xList:
+            for value in range(len(element)):
+                element[value] = int(element[value])
+
+        # Same for the y coordinates
+        for element in range(len(y)):
```
Comment (licensedtroubleshooter) [inline ExtractMarkupTool.py:58]: You are doing the same thing twice, for x and y. How about create a specific method for it?

File: ExtractMarkupTool.py [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        tuple = (xList[value], yList[value])
+        polygon.append(tuple)
+    return polygon
+
+
+def getMask(images, picture, original):
+    # Returns the mask of the polygon
+    polygon = makePolygon(images[1][picture], images[2][picture])
+    mask = Image.new('RGB', original.size, color=(255, 255, 255))
+    mask_draw = ImageDraw.Draw(mask)
```
Comment (licensedtroubleshooter) [inline ExtractMarkupTool.py]: are these calls (and one below) necessary? cuz you are returning mask, and for me it seems like you are modifying it in lines 82 and 83

File: ExtractMarkupTool.py [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+    mask_draw.polygon(polygon, fill=(0, 0, 0))
+    return mask
+
+
+def cropPictures(images, pathOriginal, pathSave):
+    # Returns the image contained in the region
```
Comment (licensedtroubleshooter) [inline ExtractMarkupTool.py]: > Returns
But I don't see a return statement. where is a mistake?

File: ImageToCSVTool.py [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+                if column==0 & line==0:
+                    pixelString=pixelString + str(pixelValue)
+                else:
+                    pixelString=pixelString + " " + str(pixelValue)
+        pixelStringVec.append(pixelString)
+    df = pd.DataFrame({'image':nameVec, 'format':formatVec, 'width':widthVec, 'height':heightVec, 'values':pixelStringVec}, columns=columns)
+    print(df)
+    df.to_csv(path+"/images.csv")
+
+path='C:/Users/Vladimir/Desktop/Mickey Dataset/V1/Test2'
```
Comment (licensedtroubleshooter) [inline ImageToCSVTool.py]: What is this?

File: ImageToCSVTool.py [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+imagesToCSV(path)
```
Comment (licensedtroubleshooter) [inline ImageToCSVTool.py]: And this?

File: ImageToCSVTool.py [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+    formatVec=[]
+    widthVec=[]
+    heightVec=[]
+    pixelStringVec=[]
+    for file in os.listdir(path):
+        nameVec.append(file[:-4])
+        formatVec.append(file[-3:])
+        width=0
+        height=0
+        print(file)
```
Comment (licensedtroubleshooter) [inline ImageToCSVTool.py]: Does it provide any useful functionality?

File: featureExtraction.py [hunk 14]
```diff
@@ -0,0 +1,43 @@
+import cv2
```
Comment (licensedtroubleshooter) [inline featureExtraction.py]: Name of the file should start with capital letter

File: featureExtraction.py [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+
+
+class FeatureExtractionTool:
+
+    def edging(self, path, imName):
+    # Returns the contours of an image
+        img = cv2.imread(path)
+        img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        edges = cv2.Canny(img, 200, 255, 3)
+        return(edges)
```
Comment (licensedtroubleshooter) [inline featureExtraction.py]: Parenthesis are redundant here

File: featureExtraction.py [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+        return greyscl
+
+    def pair(self, path, imName):
+    # Returns a paired image. Paired image should have the same name as the original image
+        pairPath=raw_input("Enter the path of the other image folder : ")
+        return cv2.imread(pairPath+"/"+imName)
+
+
+    def featureExtraction(self, function, path):
+    # Puts the images in the format required for pix2pix
```
Comment (licensedtroubleshooter) [inline featureExtraction.py]: Could you please describe this format a little?

**Review by licensedtroubleshooter — COMMENTED** [review]
In general everything is fine, but code needs a little improvement.

File: ExtractMarkupTool.py (hunk 9 above)
Comment (PierreMarTich) [reply ExtractMarkupTool.py]: These calls are necessary because mask belongs to the Image class whereas mask_draw belongs to the Draw class.

Image allows to read and save images while Draw is used to draw on it

File: ExtractMarkupTool.py [hunk 17]
```diff
@@ -2,43 +2,50 @@
...
+class ExtractMarkupTool(object):
     CSV_COLUMN_INDEX_IMAGE_NAME = 0
     CSV_COLUMN_INDEX_REGION_SHAPE_ATTRIBUTES = 5
 
     def __init__(self, csv_path):
         self.csv_path = csv_path
 
+    #Removes the characters that are not used afterwards
+    #Argument x_y stands for a charecter (x or y) that will be deleted
+    def removeChar(self, element, x_y):
```
Comment (licensedtroubleshooter) [inline ExtractMarkupTool.py]: Are you sure about this function being so simple? I mean, I remember that it was different characters being deleted for x and y.

File: CollectTool.py [hunk 18]
```diff
@@ -1,30 +1,31 @@
 import cv2
```
Comment (licensedtroubleshooter) [inline CollectTool.py]: I don't think that you've tested the code before pushing it to the repository. 
That is extremely bad habit.

File: FeatureExtraction.py [hunk 19]
```diff
@@ -1,35 +1,34 @@
...
-from PIL import Image, ImageFilter, ImageOps, ImageDraw
+from PIL import Image
 
 
 class FeatureExtractionTool:
 
-    def edging(self, path, imName):
     # Returns the contours of an image
-        img = cv2.imread(path)
+    def edges(self, path, imName):
```
Comment (licensedtroubleshooter) [inline FeatureExtraction.py]: I would name function like "getEdges" for future reader's better understanding

File: FeatureExtraction.py [hunk 20]
```diff
@@ -1,35 +1,34 @@
...
         img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
         edges = cv2.Canny(img, 200, 255, 3)
-        return(edges)
+        return edges
 
-    def greyscale(self, path, imName):
     # Returns an image in greyscale format
-        img=cv2.imread(path)
+    def greyscale(self, path, imName):
+        img=cv2.imread(path+'/'+ imName)
```
Comment (licensedtroubleshooter) [inline FeatureExtraction.py]: Binary operators

File: FeatureExtraction.py [hunk 21]
```diff
@@ -1,35 +1,34 @@
...
         greyscl = cv2.cvtColor(img, cv2.COLOR_RGB2GRAY)
         greyscl = cv2.bitwise_not(greyscl)
         return greyscl
 
-    def pair(self, path, imName):
     # Returns a paired image. Paired image should have the same name as the original image
-        pairPath=raw_input("Enter the path of the other image folder : ")
+    # This can be useful for example if we want to do "Day to Night image translation"
+    def pair(self, path, imName):
```
Comment (licensedtroubleshooter) [inline FeatureExtraction.py]: Please, think about more descriptive name of a function

File: ImageToCSVTool.py [hunk 22]
```diff
@@ -4,39 +4,38 @@
...
+                for column in range(len(imgarray[line])):
+                    pixelValue = imgarray[line][column]
+                    if column == 0 & line == 0:
+                        pixelString=pixelString + str(pixelValue)
+                    else:
+                        pixelString=pixelString + " " + str(pixelValue)
+            pixelStringVec.append(pixelString)
+        df = pd.DataFrame({'image':nameVec, 'format':formatVec, 'width':widthVec, 'height':heightVec, 'values':pixelStringVec}, columns=columns)
+        print(df)
+        df.to_csv(path+"/images.csv")
```
Comment (licensedtroubleshooter) [inline ImageToCSVTool.py]: Maybe we can ask user for the filename of csv file?

File: CollectTool.py (hunk 18 above)
Comment (PierreMarTich) [reply CollectTool.py]: The code works on my machine

File: ExtractMarkupTool.py [hunk 23]
```diff
@@ -9,16 +9,15 @@ class ExtractMarkupTool(object):
...
-    #Removes the characters that are not used afterwards
-    #Argument x_y stands for a charecter (x or y) that will be deleted
+    #Removes the characters in the csv file that are not used afterwards
+    #Argument x_y stands for which coordinate (x or y) will be also removed
     def removeChar(self, element, x_y):
         element = element.replace('_%s":' % x_y, "")
         element = element.replace(',"', "")
         element = element.replace('[', "")
         element = element.replace(']', "")
         return element
```
Comment (licensedtroubleshooter) [inline ExtractMarkupTool.py]: You didn't test your code again.

**Review by licensedtroubleshooter — APPROVED** [review]
